packages/adapter-node/index.js

```diff
@@ -16,103 +7,55 @@ import zlib from 'zlib';
 
 const pipe = promisify(pipeline);
 
+const files = fileURLToPath(new URL('./files', import.meta.url));
+
 /**
  * @typedef {import('esbuild').BuildOptions} BuildOptions
  */
 
 /** @type {import('.')} */
 export default function ({
-	entryPoint = '.svelte-kit/node/index.js',
 	out = 'build',
 	precompress,
-	env: { path: path_env = 'SOCKET_PATH', host: host_env = 'HOST', port: port_env = 'PORT' } = {},
-	esbuild: esbuild_config
+	env: { path: path_env = 'SOCKET_PATH', host: host_env = 'HOST', port: port_env = 'PORT' } = {}
 } = {}) {
 	return {
 		name: '@sveltejs/adapter-node',
 
-		async adapt({ utils, config }) {
-			utils.rimraf(out);
+		async adapt(builder) {
+			builder.rimraf(out);
 
-			utils.log.minor('Copying assets');
-			const static_directory = join(out, 'assets');
-			utils.copy_client_files(static_directory);
-			utils.copy_static_files(static_directory);
+			builder.log.minor('Copying assets');
+			builder.writeClient(`${out}/client`);
+			builder.writeServer(`${out}/server`);
+			builder.writeStatic(`${out}/static`);
 
-			if (precompress) {
-				utils.log.minor('Compressing assets');
-				await compress(static_directory);
-			}
+			builder.log.minor('Prerendering static pages');
+			await builder.prerender({
+				dest: `${out}/prerendered`
+			});
 
-			utils.log.minor('Building SvelteKit middleware');
-			const files = fileURLToPath(new URL('./files', import.meta.url));
-			utils.copy(files, '.svelte-kit/node');
 			writeFileSync(
-				'.svelte-kit/node/env.js',
-				`export const path = process.env[${JSON.stringify(
-					path_env
-				)}] || false;\nexport const host = process.env[${JSON.stringify(
-					host_env
-				)}] || '0.0.0.0';\nexport const port = process.env[${JSON.stringify(
-					port_env
-				)}] || (!path && 3000);`
+				`${out}/manifest.js`,
+				`export const manifest = ${builder.generateManifest({
+					relativePath: './server'
+				})};\n`
 			);
 
-			/** @type {BuildOptions} */
-			const defaultOptions = {
-				entryPoints: ['.svelte-kit/node/middlewares.js'],
-				outfile: join(out, 'middlewares.js'),
-				bundle: true,
-				external: Object.keys(JSON.parse(readFileSync('package.json', 'utf8')).dependencies || {}),
-				format: 'esm',
-				platform: 'node',
-				target: 'node14',
-				inject: [join(files, 'shims.js')],
-				define: {
-					APP_DIR: `"/${config.kit.appDir}/"`
+			builder.copy(files, out, {
+				replace: {
+					APP: './server/app.js',
+					MANIFEST: './manifest.js',
```

Comment:
should the adapter have to know about these? maybe we could use Vite aliases?

Reply:
Vite isn't involved here